code/exercises/affine-cipher/example.scm

```diff
@@ -0,0 +1,89 @@
+(import (rnrs))
+
+
+(define (encode key text)
+  ((get-translator key) (string-downcase text) 'encode))
+
+(define (decode key text)
+  ((get-translator key) text 'decode))
+
+;;; Private
+
+(define (atbash-format lst)
+  (let* ((chunks (windows lst 5))
+         (l (cons (car chunks)
+                  (map (lambda (chars) (cons #\space chars))
+                       (cdr chunks)))))
+    (list->string (fold-left append '() l))))
+
+(define (get-translator key)
+   (translator-factory
+    (cdr (assoc 'a key))
+    (cdr (assoc 'b key))))
+
+;; Work Horse - return a translator tuned to a specific affine key
+(define (translator-factory a b)
+  (when (not (coprime? a 26))
+    (error 'affine-cipher "a and m must be coprime" a))
+  (let* (;; generate a char pair from an alphabetic index
+         (index->pair
+          (lambda (idx)            ;; (a * x + b) % m
+            (let ((cipher-idx (modulo (+ (* a idx) b) 26)))
+              `(,(idx->char idx) . ,(idx->char cipher-idx)))))
+         ;; build an alist mapping the alphabet to the 'cipherbet'
+         (encode-map (map index->pair (iota 26)))
+         ;; build an alist mapping the 'cipherbet' to the alphabet
+         (decode-map (map riap encode-map))
+         ;; en/decrypt text and return a char list
+         (translate
+          (lambda (text tr-map)
+            (filter
+             ;; identity - to remove #f from the list
+             (lambda (x) x)
+             (map (lambda (c)
+                    (cond ((assoc c tr-map) => cdr)
+                          ((char-numeric? c) c)
+                          (else #f)))
```

Comment:
I liked the aesthetic look of this, but it should probably check if it is a number first, so it doesn't need to go through the whole alist first.  For the tests performance is not really critical, so I suppose it is fine.